cmake/iOS.cmake

```diff
@@ -159,13 +159,14 @@ set (CMAKE_OSX_SYSROOT ${CMAKE_IOS_SDK_ROOT} CACHE PATH "Sysroot used for iOS su
 
 # set the architecture for iOS 
 if (IOS_PLATFORM STREQUAL "OS")
-    set (IOS_ARCH "armv7;armv7s;arm64")
+    set (DEFAULT_IOS_ARCH "armv7;armv7s;arm64")
 elseif (IOS_PLATFORM STREQUAL "SIMULATOR")
-    set (IOS_ARCH "i386;x86_64")
+    set (DEFAULT_IOS_ARCH "i386;x86_64")
 elseif (IOS_PLATFORM STREQUAL "WATCHOS")
-    set (IOS_ARCH "armv7k")
+    set (DEFAULT_IOS_ARCH "armv7k")
```

Comment:
Would you also like to make `IOS_PLATFORM` a build option? Did you also miss an `else` here for fatal errors?

Reply:
You don't have to specify `IOS_PLATFORM` explicitly, the default one is `OS`.

Reply:
What I meant is that `IOS_PLATFORM` may be set with a typo (like `QATCHOS`). What would happen in this case?

(I realized that you did not introduce these in this PR, so I think it's fine to leave it in this PR.)